scapy/sessions.py

```diff
@@ -108,21 +136,27 @@ def _ip_process_packet(self, packet):
                 try:
                     if self.fragments[uniq][0].frag == 0:
                         # Has first fragment (otherwise ignore)
-                        defrag, missfrag = [], []
-                        _defrag_list(self.fragments[uniq], defrag, missfrag)
+                        defrag = []  # type: List[Packet]
+                        _defrag_list(self.fragments[uniq], defrag, [])
                         defragmented_packet = defrag[0]
                         defragmented_packet = defragmented_packet.__class__(
                             raw(defragmented_packet)
                         )
                         return defragmented_packet
                 finally:
                     del self.fragments[uniq]
+            return None
         else:
             return packet
 
     def on_packet_received(self, pkt):
-        pkt = self._ip_process_packet(pkt)
-        DefaultSession.on_packet_received(self, pkt)
+        # type: (Optional[Packet]) -> None
+        if not pkt:
```

Comment:
This could break when the returned packets didn't match the first filters (e.g. IP layer missing)